scripts/checkpatch_inc.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+CHECKPATCH=checkpatch.pl
+# checkpatch.pl will ignore the following paths
+CHECKPATCH_IGNORE=$(echo core/lib/lib{fdt,tomcrypt} lib/lib{png,utils,zlib})
+_CP_EXCL=$(for p in $CHECKPATCH_IGNORE; do echo ":(exclude)$p" ; done)
+
+function _checkpatch() {
+		$CHECKPATCH --quiet --ignore FILE_PATH_CHANGES \
+				--ignore GERRIT_CHANGE_ID --no-tree -
+}
+
+function checkpatch() {
+		echo "Checking commit:  "
+		git show --oneline --no-patch $1
+		git format-patch -1 $1 --stdout -- $_CP_EXCL . | _checkpatch
+}
+
+function checkstaging() {
+		echo "Checking staging area:  "
+		git diff --cached . | _checkpatch
+}
+
+function checkdiff() {
+		echo "Checking squashed commits (diff $1...$2)"
```

Comment:
IMO, "squashed" is not very appropriate here -- it is mostly OK for Travis because we're checking a pull request that has multiple patches, some of them are review comments are will be squashed ultimately.
'Checking diff (diff $1...$2):' maybe? Any better suggestion?

We could as well leave the messages out of the functions, then Travis and this script could print whatever they want. Well that's really nitpicking so I leave it up to you, feel free to keep this unchanged if you wish :)